src/mlpack/bindings/python/tests/test_python_binding.py

```diff
@@ -140,6 +140,135 @@ def testNumpyMatrixForceCopy(self):
+      self.assertEqual(2 * x[j, 2], output['matrix_out'][j, 2])
+
+  def testNumpyFContiguousMatrixForceCopy(self):
+    """
+    The matrix with F_CONTIGUOUS set we pass in, we should get back with the third 
+    dimension doubled and the fifth forgotten.
+    """
+    x = np.array(np.random.rand(100, 5),order = 'F');
+
+    output = test_python_binding(string_in='hello',
+                                 int_in=12,
+                                 double_in=4.0,
+                                 matrix_in=x,
+                                 copy_all_inputs=True)
+
+    self.assertEqual(output['matrix_out'].shape[0], 100)
+    self.assertEqual(output['matrix_out'].shape[1], 4)
+    self.assertEqual(output['matrix_out'].dtype, np.double)
+    for i in [0, 1, 3]:
+      for j in range(100):
+        self.assertEqual(x[j, i], output['matrix_out'][j, i])
+
+    for j in range(100):
+      self.assertEqual(2 * x[j, 2], output['matrix_out'][j, 2])
+
+  def testPandasDataFrameMatrix(self):
+    """
+    The matrix we pass in, we should get back with the third dimension doubled
+    and the fifth forgotten.
+    """
+    x = pd.DataFrame(np.random.rand(100, 5))
+    z = copy.copy(x)
+
+    output = test_python_binding(string_in='hello',
+                                 int_in=12,
+                                 double_in=4.0,
+                                 matrix_in=z)
+
+    self.assertEqual(output['matrix_out'].shape[0], 100)
+    self.assertEqual(output['matrix_out'].shape[1], 4)
+    self.assertEqual(output['matrix_out'].dtype, np.double)
+    for i in [0, 1, 3]:
+      for j in range(100):
+        self.assertEqual(x.iloc[j, i], output['matrix_out'][j, i])
+
+    for j in range(100):
+      self.assertEqual(2 * x.iloc[j, 2], output['matrix_out'][j, 2])
+
+  def testPandasDataFrameMatrixForceCopy(self):
+    """
+    The matrix we pass in, we should get back with the third dimension doubled
+    and the fifth forgotten.
+    """
+    x = pd.DataFrame(np.random.rand(100, 5))
+
+    output = test_python_binding(string_in='hello',
+                                 int_in=12,
+                                 double_in=4.0,
+                                 matrix_in=x,
+				 copy_all_inputs=True)
+
+    self.assertEqual(output['matrix_out'].shape[0], 100)
+    self.assertEqual(output['matrix_out'].shape[1], 4)
+    self.assertEqual(output['matrix_out'].dtype, np.double)
+    for i in [0, 1, 3]:
+      for j in range(100):
+        self.assertEqual(x.iloc[j, i], output['matrix_out'][j, i])
+
+    for j in range(100):
+      self.assertEqual(2 * x.iloc[j, 2], output['matrix_out'][j, 2])
+
+  def testPandasSeries(self):
+    """
+    Test a Pandas Series input paramter
+    """
+    x =  pd.Series(np.random.rand(100))
+    z = copy.copy(x)
+
+    output = test_python_binding(string_in='hello',
+                                 int_in=12,
+                                 double_in=4.0,
+                                 col_in=z)
```

Comment:
We should test a Pandas series also to be passed as `matrix_in`.  But that may require the fix for one-dimensional objects which is another part of #1710.  If so, we can wait to add that.

Reply:
If you can let me know what you think here before we merge, I'd appreciate that. :+1:

Reply:
hi @rcurtin when we pass Pandas Series as `matrix_in` it gives Segmentation Fault as I believe this is because the function `numpy_to_mat_d()`, expects a `numpy.ndarray[numpy.double_t, ndim=2]`, but we are passing one with `ndim=1`

Reply:
Ok, that's what I figured but I just wanted to check.  We can handle it in a different PR then. :)

Reply:
Yes we can test this on different PR :)